homework/src/index.js

```diff
@@ -30,17 +33,108 @@
-      }
+  function renderRepository(repository, container) {
+    container.innerHTML = '';
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository: ',
+    });
+    const td2 = createAndAppend('td', tr1);
+    createAndAppend('a', td2, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, {
+      text: 'Description:'
+    });
+    createAndAppend('td', tr2, {
+      text: repository.description
+    });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, {
+      text: 'Forks:'
+    });
+    createAndAppend('td', tr3, {
+      text: repository.forks
+    });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, {
+      text: 'Updated:'
+    });
+    createAndAppend('td', tr4, {
+      text: repository.updated_at
+    });
+  }
+  function renderContributors(contributors, container) {
+    container.innerHTML = '';
+    const h2 = createAndAppend('h2', container, {
+      text: 'Contributors'
+    });
+    const ul = createAndAppend('ul', container);
+
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url
+      });
+      createAndAppend('div', li, {
+        text: contributor.contributions
+      });
     });
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    fetchJSON(url)
+      .then(repositories => {
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        console.log(repositories);
+        const header = createAndAppend('header', root);
+        const h1 = createAndAppend('h2', header, {
+          text: 'HYF Repositories'
+        });
+        const select = createAndAppend('select', header);
+        const leftHandContainer = createAndAppend('div', root, {
+          id: 'left-hand'
+        });
+        const rightHandContainer = createAndAppend('div', root, {
+          id: 'right-hand'
+        });
+        repositories.forEach((repository, index) => {
+          createAndAppend('option', select, {
+            text: repository.name,
+            value: index
+          });
+        });
+        select.addEventListener('change', () => {
+          const index = select.value;
+          const repository = repositories[index];
+          renderRepository(repository, leftHandContainer);
+          fetchJSON(repository.contributors_url)
+            .then(contributors => {
+              renderContributors(contributors, rightHandContainer);
+            })
+            .catch(err => err.message);
```

Comment:
You are not rendering the error to the DOM. When an error occurs there is no way for a user to know what happened. In case of an error you should, for instance, create a `div`, set its text content to the error message string and append the `div` as a child to the `root` div. Look as line 136 where this is done.